lib/puppet/settings.rb

```diff
@@ -149,6 +149,7 @@ def initialize
 
     @hooks_to_call_on_application_initialization = []
     @deprecated_setting_names = []
+    @mutable_settings_in_context = %i{tasks strict strict_variables code}
```

Comment:
I don't necessarily disagree about only doing this for certain settings, but what happens if someone starts using a new setting in the bad way? I don't think I see anything in here that prevents them from doing that, but maybe I'm missing something?

Reply:
No, there’s nothing here to do that. I would love to see the rest of the settings made immutable, but I didn’t think that it was part of this work. (Maybe that’s part of what I’m forgetting??)

Reply:
No, there’s nothing here to do that. I would love to see the rest of the settings made immutable, but I didn’t think that it was part of this work. (Maybe that’s part of what I’m forgetting??)

Reply:
Yeah it's probably not. Something to think about for the future.

Reply:
I think we scoped that out when we talked about file serving. There's a setting that explicitly controls how often Puppet will re-read config files that needs to be deprecated.